### PR TITLE
Josue/orb updates

### PR DESCRIPTION
**Pull request by JosueMolinaMorales**
This PR completes part of [8276](https://app.shortcut.com/clarkcan/story/8276/implement-circleci-orb) and [8647](https://app.shortcut.com/clarkcan/story/8647/implement-command-tag-repo)

For tagging, i believe no authentication is necessary because the -checkout command would be use before the tag-repo command would be use and looking at the circleci docs about the -checkout command this is what i saw about authentication : `This command will automatically add the required authenticity keys for interacting with GitHub and Bitbucket over SSH`

### CODE REVIEW

**Review by StarChaser44 — DISMISSED** [review]

**Review by EVogel1999 — DISMISSED** [review]

File: src/commands/tag-repo.yml [hunk 1]
```diff
@@ -2,7 +2,23 @@ description: >
...
-      command: echo "Not Yet Implemented"
+    name: Validate params
+    command: |
+      if [[ -z "<<parameters.tag>>"]]; then
+        echo "tag parameter must be defined"
+        exit 1
+      fi
+  - run:
+    name: Tag Repo
+    command: |
```
Comment (Cwagne17) [inline src/commands/tag-repo.yml:21]: use >

**Review by Cwagne17 — CHANGES_REQUESTED** [review]
Other the lint requests. Check where the status are failing.

**Review by EVogel1999 — APPROVED** [review]

File: src/commands/setup-aws.yml [hunk 2]
```diff
@@ -3,23 +3,29 @@ description: >
   This command should be used prior to working with aws orbs or cli commands.
```
Comment (Cwagne17) [inline src/commands/setup-aws.yml:3]: We can remove this file with the new insight I posted on Andy's story.

File: src/commands/setup-docker.yml [hunk 3]
```diff
@@ -18,5 +18,7 @@ parameters:
 steps:
   - setup_remote_docker
   - docker/install-docker
-  - run: |
-    docker login -u << parameters.docker-username >> -p << parameters.docker-password >>
\ No newline at end of file
+  - run:
```
Comment (Cwagne17) [inline src/commands/setup-docker.yml:21]: use docker/check not docker cli

File: src/commands/tag-repo.yml [hunk 4]
```diff
@@ -2,7 +2,22 @@ description: >
   This command will tag the github repository with the semantic version of the project.
   This command should be used prior to deploying the project.
 
+parameters:
+  tag:
+    type: string
+    description: |
+      Tag being assigned
```
Comment (Cwagne17) [inline src/commands/tag-repo.yml:9]:
```suggestion
      The semantic version of the project trying to be deployed
      https://semver.org/spec/v2.0.0.html
```

File: src/commands/tag-repo.yml [hunk 5]
```diff
@@ -2,7 +2,22 @@ description: >
...
-      name: "Not Yet Implemented"
-      command: echo "Not Yet Implemented"
+      name: Validate params
+      command: >
+        if [[ -z "<<parameters.tag>>"]]; then
+          echo "tag parameter must be defined"
+          exit 1
+        fi
+  - run:
+      name: Tag Repo
```
Comment (Cwagne17) [inline src/commands/tag-repo.yml:20]:
```suggestion
      name: "Tag $CIRCLE_PROJECT_REPONAME Repository"
```

Comment (Cwagne17) [reply src/commands/tag-repo.yml:20]: https://circleci.com/docs/2.0/env-vars/?utm_source=google&utm_medium=sem&utm_campaign=sem-google-dg--uscan-en-dsa-maxConv-auth-brand&utm_term=g_-_c__dsa_&utm_content=&gclid=Cj0KCQjwr-SSBhC9ARIsANhzu14qQHzsiV1UDPIUlBqyEjQUJqF-y_q74BFkorF2NMYyJ8SI6lYLBR4aAvgoEALw_wcB#built-in-environment-variables

**Review by Cwagne17 — CHANGES_REQUESTED** [review]
Small requests. Looks good!

**Review by pzalep1 — COMMENTED** [review]
After Chris's comments are addressed this looks good!

**Review by StarChaser44 — COMMENTED** [review]
Build is failing

**Comment by JosueMolinaMorales** [comment]
> Build is failing

Builds are going to fail since not all job implementations are in main.

File: .circleci/config.yml [hunk 6]
```diff
@@ -51,5 +52,10 @@ workflows:
           vcs-type: << pipeline.project.type >>
           pipeline-number: << pipeline.number >>
           filters:
+            branches:
```
Comment (Cwagne17) [inline .circleci/config.yml:55]: lets remove the filter then remove the test-deploy jobs and commands being tested

File: src/jobs/deploy-s3.yml [hunk 7]
```diff
@@ -1,9 +1,80 @@
-description: >
+description: |
   This job deploys an application to aws s3.
 
 executor: node/default
 
+parameters:
+#required parameters
```
Comment (Cwagne17) [inline src/jobs/deploy-s3.yml:7]:
```suggestion
 # Required Parameters
```

File: src/jobs/deploy-s3.yml [hunk 8]
```diff
@@ -1,9 +1,80 @@
...
+  tag:
+    type: string
+    description: |
+      The semantic version of the project trying to be deployed
+      https://semver.org/spec/v2.0.0.html
+#optional parameters with defaults
```
Comment (Cwagne17) [inline src/jobs/deploy-s3.yml:13]:
```suggestion
 # Not Required Parameters
```

File: src/jobs/deploy-s3.yml [hunk 9]
```diff
@@ -1,9 +1,80 @@
...
+  s3-bucket:
+    type: env_var_name
+    default: S3_BUCKET
+    description: |
+      AWS S3 Bucket URI used to identify the resource
+
```
Comment (Cwagne17) [inline src/jobs/deploy-s3.yml:19]: can we remove these spaces

File: src/jobs/deploy-s3.yml [hunk 10]
```diff
@@ -1,9 +1,80 @@
...
 steps:
+  - checkout
+  - tag-repo:
+      tag: <<parameters.tag>>
+  - run:
+      name: Install dependencies
+      command: npm install
+  - run:
+      name: Build Application
+      command: npm run build --aot --build-optimizer=true --optimization=true --configuration=production
```
Comment (Cwagne17) [inline src/jobs/deploy-s3.yml:63]: we should change this to ng build 
```suggestion
      command: ng build --aot --build-optimizer=true --optimization=true --configuration=production
```

**Review by Cwagne17 — CHANGES_REQUESTED** [review]

**Review by StarChaser44 — COMMENTED** [review]
Chris has some comments

**Review by pzalep1 — COMMENTED** [review]
Once Chris's comments are addressed this will be good to go!

**Review by EVogel1999 — COMMENTED** [review]
After everyone's comments are addressed, looks good!

**Review by Cwagne17 — APPROVED** [review]

**Review by EVogel1999 — DISMISSED** [review]

**Review by pzalep1 — DISMISSED** [review]

**Review by Cwagne17 — APPROVED** [review]

**Review by AStoerkel1 — APPROVED** [review]

**Pull request merged**